src/node/SetupServerApi.ts

```diff
@@ -107,7 +150,29 @@ export class SetupServerApi
     )
   }
 
+  /**
+   * Wraps the given function in a boundary. Any changes to the
+   * network behavior (e.g. adding runtime request handlers via
+   * `server.use()`) will be scoped to this boundary only.
+   * @param callback A function to run (e.g. a test)
+   */
+  public boundary<Fn extends (...args: Array<unknown>) => unknown>(
```

Comment:
neat!  

should every call to `use` create some kind of `boundary` internally - can we avoid leaking that to the end user? (sorry if this is still early)

Reply:
Yes and no. We can use `.enterWith()` from `AsyncLocalStorage` on each `server.use()`, and while it would solve _some_ use cases, it would break the other. 

The problem is that relying on async storage internally makes it hard to support sequential tests. Here's an example:

```js
const server = setupServer()

beforeAll(() => {
  server.use(...someHandlers)
  server.listen()
})

it('test', () => {
  server.use(...anotherHandlers)
})
```

The issue here is that the context created by the `server.use()` call in `beforeAll` will be different and inaccessible by the context created by `sever.use()` in the test ( they have different execution async ids). 

> I also have concerns that using `.enterWith()` on each `server.use()` call creates nested execution contexts (imagine you have multiple `.use()` calls in a single test), which may have negative impact on runtime performance. 

It is for this reason I'm proposing an explicit `server.boundary()` API.

Reply:
that makes sense.  it's too bad that this leaks though, but I don't think that's a big deal.

Reply:
I actually think this is very good that it leaks. It makes a server boundary a conscious choice. Just as running tests in parallel is a conscious choice (no testing framework does it by default). 

Server boundary is also incredibly powerful outside of testing too! Imagine you want to quickly fail all requests coming from a sub-tree in your app:

```js
const whateverRenderReturns = server.boundary(() => {
  http.all('*', () => HttpResponse.error())
  return render(<Subtree />)
})
```